### ISSUE TITLE
oxlint: `overrides[].files` globs with extglob `?()` optional groups silently don't match

### ISSUE THREAD

**Issue by mymx2** [oxc-project/oxc#25134]
## Summary

When an `overrides[].files` glob uses extglob optional groups like `?()` (e.g. `**/*.test.?([cm])[jt]s?(x)`), the pattern silently matches nothing, so the whole override block is never applied. There is no error or warning, which makes this very hard to notice: scoped rules either don't run at all, or run at their category-default severity instead of the configured one.

Brace expansion `{a,b}` works fine, so the failure is specific to extglob operators like `?()`.

## Reproduction

`.oxlintrc.json`:

```json
{
  "plugins": ["vitest"],
  "rules": {
    "vitest/no-identical-title": "error"
  },
  "overrides": [
    {
      "files": ["**/*.test.?([cm])[jt]s?(x)"],
      "rules": {
        "vitest/no-focused-tests": "error",
        "vitest/consistent-test-it": ["error", { "fn": "it" }]
      }
    }
  ]
}
```

`foo.test.ts`:

```ts
import { expect, it, test } from 'vitest'

test('does something', () => {
  expect(true).toBe(true)
})

test.only('focused', () => {
  expect(true).toBe(true)
})
```

## Actual behavior

With `files: ["**/*.test.?([cm])[jt]s?(x)"]`:

```
Found 2 warnings and 0 errors.
```

- `vitest/consistent-test-it` from the override never runs (silent).
- `vitest/no-focused-tests` reports at `warn` instead of the configured `error` — i.e. the override's severity is ignored.

## Expected behavior

Same as with the plain glob `files: ["**/*.test.ts"]`, which correctly reports:

```
Found 1 warning and 3 errors.
```

(both override rules fire at `error`)

Brace expansion also works: `files: ["**/*.test.{js,jsx,ts,tsx,cjs,mjs,cts,mts}"]` produces the expected 3 errors.

## Why this matters

`?()` extglob patterns are common in the ESLint ecosystem — for example `antfu/eslint-config` uses `GLOB_TESTS = ['**/*.test.?([cm])[jt]s?(x)', ...]`, so configs ported over from ESLint silently lose their scoped rules. Two things would help:

1. Support extglob operators (`?()`, `*()`, `+()`, `@()`, `!()`) in override file matching, or
2. If unsupported, emit a config error/warning for patterns that can't be compiled — a silent no-match turns any ported config into a false sense of security.

## Environment

- oxlint 1.76.0 (Windows x64)
- Reproduced both via raw `oxlint` CLI and through Vite+ (`vp lint`) config mapping.
